### ISSUE TITLE
Proposal: add `-fshort-enums` support to translate-c

### ISSUE THREAD

**Issue by ehaas** [ziglang/zig#8662]
GCC & Clang have a command-line option `-fshort-enums` that can be used to control how many bytes are allocated for an enum type. It's equivalent to putting `__attribute__((packed))` on all enums. From the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc-11.1.0/gcc/Structures-unions-enumerations-and-bit-fields-implementation.html#Structures-unions-enumerations-and-bit-fields-implementation): 

> Normally, the type is `unsigned int` if there are no negative values in the enumeration, otherwise `int`. If `-fshort-enums` is specified, then if there are negative values it is the first of `signed char`, `short` and `int` that can represent all the values, otherwise it is the first of `unsigned char`, `unsigned short` and `unsigned int` that can represent all the values."

Super basic demonstration, compare output of the following C program when compiled with `-fshort-enums` vs not:
```c
#include <stdio.h>
typedef enum { FOO } MyEnum;
int main(void) {
    printf("%d\n", (int)sizeof(MyEnum));
    return 0;
}
```

The relevance to translate-c is that enum sizes affect ABI compatibility. Currently translate-c always translates enums with an underlying type of `c_int` or `c_uint`, but this won't work if linking with code that was built with `-fshort-enums`. Short enums are fairly common in embedded systems where space is at a premium.

One way to solve this would be to add a special preprocessor symbol like `__ZIG_SHORT_ENUMS` (which can be used with `zig translate-c` via `-D__ZIG_SHORT_ENUMS`, and can be used with `@cImport` via `@cDefine("__ZIG_SHORT_ENUMS", {});`. translate-c could look for the existence of that symbol and use it to control whether enums are generated with the short enum algorithm described above. One downside to this approach is that it reifies the short enum decision into the generated code, so it wouldn't be possible to translate code once and have it automatically work across platforms.

**Comment by andrewrk** [comment]
Not only for translate-c but I think we should also add support for it in the main CLI options as well, but make it clear that it applies only to compiled C code. I think if you go to implement this it will become clear that these features are intertwined.

**Comment by ehaas** [comment]
Are you suggesting that instead of doing this in translate-c; have a CLI option `-fshort-enums` that controls the size of extern enums?

**Comment by andrewrk** [comment]
Wait a minute, sorry, I just realized this should already be working, like this:

```
./zig translate-c test.c  -cflags -fshort-enums -- -lc
```

I tried it just now and it did not affect the output, but that seems like a bug. Not sure why that is happening.

Semi-related: 557eb414eee96201d1ae01b15ebf8955eca1da6d

**Comment by ehaas** [comment]
Ah ok, it might just be a bug in the cli parsing then. I just tested it out and `-fshort-enums` didn't make it into the clang args. But it's probably a much easier fix to do that than to do the preprocessor thing I suggested, I can take a look at it.

**Comment by ehaas** [comment]
Closed in favor of #8676
